src/swap/SwapScreen.tsx

```diff
@@ -202,6 +203,10 @@ export function SwapScreen({ route }: Props) {
   const networkFeeInfoBottomSheetRef = useRef<BottomSheetRefType>(null)
   const slippageInfoBottomSheetRef = useRef<BottomSheetRefType>(null)
 
+  const [selectingNoUsdPriceToken, setSelectingNoUsdPriceToken] = useState<TokenBalance | null>(
+    null
+  )
```

Comment:
Could we move this to the reducer? as a new event?

This way all the component state remains there and we can reset based on the events dispatched.

Otherwise it will be tempting for the next person to touch this to add another `useState` again.

Reply:
oops, yes i should have added this to the reducer. i was fixated on refactoring to move the toast into the swap screen and didn't really think further